src/udisks2block.cpp

```diff
@@ -274,9 +260,8 @@ bool UDisks2::Block::isReadOnly() const
 
 bool UDisks2::Block::isExternal() const
 {
-    const QString prefDevice = preferredDevice();
-    return prefDevice != QStringLiteral("/dev/sailfish/home") && prefDevice != QStringLiteral("/dev/sailfish/root")
-            && mountPath() != QStringLiteral("/home") && mountPath() != QStringLiteral("/");
+    return m_drive.contains(QLatin1String("Removable")) &&
+        m_drive.value(QStringLiteral("Removable")).toBool();
```

Comment:
Just wondring which is more efficient, this or `return m_drive.value(QStringLiteral("Removable"), false).toBool();`.

That said, this probably should use the same `QStringLiteral("Removable")` value for both in any case. It can be, for example, made global and placed in anonymous namespace as we regularly do for this kind of strings.

Reply:
> 

I think GCC will optimize both away in that case. Using m_drive.contains(QLatin1String("Removable")) first is more safer.

Reply:
If this was rust I would trust the compiler to optimize it, but for GCC and C++ I'm not so sure. :laughing: Anyways, it was just a thought, that latter part of the comment is more important. At least I can't see any reason to use `QLatin1String` for `contains` when `QStringLiteral` is anyway used for `value`. Also if there are more instances of "Removable" in the code, they could use that too.

Reply:
Quite unlikely that GCC will optimize anything here. But then again this shouldn't be too hot of a code path either.

Though don't see anything safer in checking the .contains() first, the value() is designed to return the fallback value if there is no given key available.